shared/src/error/mod.rs

```diff
@@ -147,6 +147,8 @@ pub enum ErrorNumber {
     Forbidden = 11,
     /// The passed buffer is too small.
     BufferTooSmall = 12,
+    /// The actor is executing in a read-only context.
+    ReadOnly = 13,
```

Comment:
I considered using an existing error but:

- IllegalOperation is ambiguous on `set_root` (returned both if the actor has already been deleted, or is in read-only mode).
- Forbidden is ambiguous on `self_destruct` if the beneficiary is self.

In general, I'm starting to regret having "common" errors and am becoming convinced we want more specific ones. We can still share errors between syscalls, but errors like `ActorDeleted` is probably better than `IllegalOperation`.